plugins/modules/device_credential_intent.py

```diff
@@ -754,6 +789,41 @@ def validate_input(self):
         self.status = "success"
         return self
 
+    def validate_api_response(self, response):
```

Comment:
Can we move this API in base class. This is the common API whereas anyone can use it